### DISCUSSION TITLE
How suitable is Ariadne for use in AWS Lambda?

### DISCUSSION THREAD

**Discussion by pclements12** [mirumee/ariadne#758] [Q&A] [answered]
I'm interested in running a single AWS Lambda function as a GraphQL entry point.

I did some googling and haven't seen a lot of examples out there using Ariadne in this way, so I'm curious if anyone's ever tried it out.

Can Ariadne be used with AWS Lambda?
If so, what are the start-up characteristics for Ariadne that might impact the performance of a Lambda function?

Any reasons to avoid this combination for some reason?

Thanks!

**Comment by howbazaar** [comment]
We are currently using ariadne in AWS lambdas.

We are currently using chalice to deploy the app, which has the lambda sitting behind an apigateway.

Our app.route just uses the graphql_sync call rather than any async. We construct the graphql schema at lambda load time, so this is done only once while the lambda is hot.

**Reply by PaulRudin** [reply]
If you have subscriptions then you'd need to be aware that the lambda instances don't last terribly long, so connections will drop from time to time. That may not matter as long as your clients are smart enough to reconnect when the connection drops. If you're not using subscriptions then it should be fine, although I don't know what the cold start performance is like. You can set provisioned concurrency - but then you're potentially paying for instances are not being used, and it might be better to use a different, long-running hosting option.

**Comment by mociepka** [comment, marked as answer]
Check this documentation about Ariadne and AWS Lambda.

https://ariadnegraphql.org/docs/aws-lambda

**Comment by pkucmus** [comment]
Check this out: https://github.com/mirumee/ariadne-lambda
